src/reflection.ts

```diff
@@ -528,14 +530,11 @@ export class Parser {
     if (fieldType === null) {
       throw new Error('Malformed schema: "type" field of Field not populated.');
     }
-    const parentIsStruct = this.getType(typeIndex).isStruct();
-    if (
-      fieldType.baseType() !== reflection.BaseType.Obj &&
-      fieldType.baseType() !== reflection.BaseType.Union
-    ) {
-      throw new Error("Field " + field.name() + " is not an object or union type.");
+    if (fieldType.baseType() !== reflection.BaseType.Obj) {
```

Comment:
Removed _union_ type support from this function because `fieldType.index()` is not correct for the semantics here for union types